### PR TITLE
Improved key generation for MacTests

### PR DESCRIPTION
**Pull request by zzambers**
Improved MacTests to handle cases, where keygen is not directly available

### CODE REVIEW

File: cryptotest/tests/MacTests.java [hunk 1]
```diff
@@ -67,23 +67,39 @@ protected void checkAlgorithm(Provider.Service service, String alias) throws Alg
         try {
             Mac md = Mac.getInstance(alias, service.getProvider());
             byte[] b = new byte[]{1, 2, 3};
+            Provider provider =  service.getProvider();
+            String algorithm = service.getAlgorithm();
+            String generatorAlgorithm;
 
-            if (service.getAlgorithm().contains("PBE")) {
+            if (algorithm.contains("PBE")) {
```
Comment (judovana) [inline cryptotest/tests/MacTests.java:74]: Why this?

Comment (zzambers) [reply cryptotest/tests/MacTests.java:74]: I don't understand the question. I introduced some variables, so I would not be copying the same expression over multiple places.

Comment (judovana) [reply cryptotest/tests/MacTests.java:74]: I do not understand the question too.

Comment (zzambers) [reply cryptotest/tests/MacTests.java:74]: :)

**Pull request merged**